packages/@react-spectrum/s2/src/RangeCalendar.tsx

```diff
@@ -10,38 +10,390 @@
-  Button,
-  CalendarCell,
+  ButtonProps,
+  CalendarCellProps,
+  CalendarCellRenderProps,
   CalendarGrid,
+  CalendarGridBody,
+  CalendarGridHeader,
+  CalendarHeaderCell,
+  ContextValue,
   DateValue,
-  Heading,
+  RangeCalendarStateContext,
   Text
 } from 'react-aria-components';
-import {ReactNode} from 'react';
+import {baseColor, focusRing, lightDark, style} from '../style' with {type: 'macro'};
+import ChevronLeftIcon from '../s2wf-icons/S2_Icon_ChevronLeft_20_N.svg';
+import ChevronRightIcon from '../s2wf-icons/S2_Icon_ChevronRight_20_N.svg';
+import {controlFont, getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+import {createContext, CSSProperties, ForwardedRef, forwardRef, Fragment, ReactNode, useContext, useMemo, useRef} from 'react';
+import {forwardRefType, ValidationResult} from '@react-types/shared';
+import {getEraFormat} from '@react-aria/calendar';
+import {useDateFormatter} from '@react-aria/i18n';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
 
 
 export interface RangeCalendarProps<T extends DateValue>
-  extends AriaRangeCalendarProps<T> {
-  errorMessage?: string
+  extends Omit<AriaRangeCalendarProps<T>, 'visibleDuration' | 'style' | 'className' | 'styles'>,
+  StyleProps {
+  errorMessage?: ReactNode | ((v: ValidationResult) => ReactNode),
+  visibleMonths?: number
 }
 
-export function RangeCalendar<T extends DateValue>(
-  {errorMessage, ...props}: RangeCalendarProps<T>
-): ReactNode {
+export const RangeCalendarContext = createContext<ContextValue<Partial<RangeCalendarProps<any>>, HTMLDivElement>>(null);
+
+
+const calendarStyles = style({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: 24,
+  width: 'fit'
+}, getAllowedOverrides());
+
+const headerStyles = style({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  width: 'full'
+});
+
+const headingStyles = style({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  margin: 0,
+  width: 'full'
+});
+
+const titleStyles = style({
+  font: 'title-lg',
+  textAlign: 'center',
+  flexGrow: 1,
+  flexShrink: 0,
+  flexBasis: '0%',
+  minWidth: 0
+});
+
+const headerCellStyles = style({
+  font: 'title-sm',
+  cursor: 'default',
+  textAlign: 'center',
+  flexGrow: 1
+});
+
+const cellStyles = style<CalendarCellRenderProps>({
+  paddingX: 4,
+  '--cell-gap': {
+    type: 'paddingStart',
+    value: 4
+  },
+  paddingY: 2
+});
+
+const cellInnerWrapperStyles = style<CalendarCellRenderProps>({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  width: 'full',
+  boxSizing: 'border-box',
+  borderStartRadius: {
+    isSelectionStart: 'full'
+  },
+  borderEndRadius: {
+    isSelectionEnd: 'full'
+  },
+  outlineStyle: 'none'
+});
+
+const innerCellStyles = style<CalendarCellRenderProps>({
+  ...focusRing(),
+  outlineOffset: {
+    default: -2,
+    isToday: 2,
+    isSelected: 2
```

Comment:
Also interesting that this is outside in the middle of a selected range in addition to the dark blue end caps